app/src/testIntegration/kotlin/io/template/data/session/SessionDataSourceImplTest.kt

```diff
@@ -42,7 +45,10 @@ class SessionDataSourceImplTest : KoinTest {
         ApplicationProvider.getApplicationContext<TemplateApp>().diDeclaration()
     )
 
-    private val securityProvider: SecurityProvider by inject()
+    private val securityProvider: SecurityProvider = mockk(relaxed = true) {
```

Comment:
that was supposed to be end-to-end integration test -> mockk breaks this. Also, tests are working fine locally for me with `by inject()`

Reply:
It fails to me, because of encryption used in the SecurityProvider

Reply:
I reverted to the main code and tests are failing:
`SessionDataSourceImplTest > session_onAppStart_isEmpty FAILED
    java.security.NoSuchProviderException at SessionDataSourceImplTest.kt:53
        Caused by: java.util.jar.JarException at SessionDataSourceImplTest.kt:53

SessionDataSourceImplTest > session_preferences_parsed_and_read FAILED
    java.security.NoSuchProviderException at SessionDataSourceImplTest.kt:68
        Caused by: java.util.jar.JarException at SessionDataSourceImplTest.kt:53

SessionDataSourceImplTest > session_preferences_parsed_and_stored FAILED
    java.security.NoSuchProviderException at SessionDataSourceImplTest.kt:58
        Caused by: java.util.jar.JarException at SessionDataSourceImplTest.kt:53
`

Reply:
encryption is mocked via DI and `test.fake.keystore` package –> so it doesn't fail for me. I'd like to understand and fix why it fails on your machine